src/backend/distributed/master/citus_tools.sql

```diff
@@ -0,0 +1,243 @@
+	    (SELECT count(*) FROM pg_dist_shard b WHERE b.logicalrelid = table2)
+	INTO colocated_shard_count, table1_shard_count, table2_shard_count
+	FROM pg_dist_shard tba JOIN pg_dist_shard tbb USING(shardminvalue, shardmaxvalue)
+	WHERE tba.logicalrelid = table1 AND tbb.logicalrelid = table2;
+
+	IF (table1_shard_count != table2_shard_count OR
+		table1_shard_count != colocated_shard_count)
+	THEN
+		RETURN false;
+	END IF;
+
+	SELECT count(*) 
+	INTO table2_placement_count
+	FROM pg_dist_shard ds JOIN pg_dist_shard_placement sp USING (shardid)
+	WHERE ds.logicalrelid = table1 AND sp.shardstate = 1;
+	
+	SELECT count(*)
+	INTO table1_placement_count
+	FROM pg_dist_shard ds JOIN pg_dist_shard_placement sp USING (shardid)
+	WHERE ds.logicalrelid = table2 AND sp.shardstate = 1;
+	
+	IF (table1_placement_count != table2_placement_count) THEN
+		RETURN false;
+	END IF;
+	
+	RETURN true;
+END;
+$function$;
+
+CREATE OR REPLACE FUNCTION citus_run_on_all_workers(command text,
+													parallel bool default true,
+													OUT nodename text,
+													OUT nodeport int,
+													OUT success bool,
+													OUT result text)
+	RETURNS SETOF record
+	LANGUAGE plpgsql
+	AS $function$
+DECLARE
+	workers text[];
+	ports int[];
+	commands text[];
+BEGIN
+	WITH citus_workers AS (
+		SELECT * FROM master_get_active_worker_nodes() ORDER BY node_name, node_port)
+	SELECT array_agg(node_name), array_agg(node_port), array_agg(command)
+	INTO workers, ports, commands
+	FROM citus_workers;
+
+	RETURN QUERY SELECT * FROM master_run_on_worker(workers, ports, commands, parallel);
+END;
+$function$;
+
+
+CREATE OR REPLACE FUNCTION citus_run_on_all_placements(table_name regclass, command text,
+													   parallel bool default true,
+													   OUT nodename text,
+													   OUT nodeport int,
+													   OUT shardid bigint,
+													   OUT success bool, OUT result text)
+	RETURNS SETOF record
+	LANGUAGE plpgsql
+	AS $function$
+DECLARE
+	workers text[];
+	ports int[];
+	shards bigint[];
+	commands text[];
+BEGIN
+	WITH citus_placements AS (
+		SELECT
+			ds.logicalrelid::regclass AS tablename,
+			ds.shardid AS shardid,
+			shard_name(ds.logicalrelid, ds.shardid) AS shardname,
+			dsp.nodename AS nodename, dsp.nodeport::int AS nodeport
+		FROM pg_dist_shard ds JOIN pg_dist_shard_placement dsp USING (shardid)
+		WHERE dsp.shardstate = 1 and ds.logicalrelid::regclass = table_name
+		ORDER BY ds.logicalrelid, ds.shardid, dsp.nodename, dsp.nodeport)
+	SELECT
+		array_agg(cp.nodename), array_agg(cp.nodeport), array_agg(cp.shardid),
+		array_agg(format(command, cp.shardname))
+	INTO workers, ports, shards, commands
+	FROM citus_placements cp;
+
+	RETURN QUERY
+		SELECT r.node_name, r.node_port, shards[ordinality],
+			r.success, r.result
+		FROM master_run_on_worker(workers, ports, commands, parallel) WITH ORDINALITY r;
+END;
+$function$;
+
+
+CREATE OR REPLACE FUNCTION citus_run_on_all_colocated_placements(table_name1 regclass,
```

Comment:
I don't agree that shipping features we know are broken is a good idea just because we plan on fixing them in the next release. As-is this has a high chance of doing something the user doesn't expect. What if they try to do a `INSERT INTO SELECT` but the source shard is inactive and has stale data while the destination shard is active: now it has bad data written to it!

How about one of these:
- Fail if a placement is marked inactive but it's matching placement isn't.
- Fail if any of the involved placements are marked inactive